pkg/config/matcher.go

```diff
@@ -28,3 +101,88 @@ func matchByAttributes(actual *processAttrs, required *Attributes) bool {
 	}
 	return true
 }
+
+func restartObjectWithAttributes(ctx context.Context, clientSet kubernetes.Interface, required *Attributes) error {
+	// pod labels is not compatible with metadata - unless it's namespace
+	// is namespace always required? does it need to be a simple string?
+	labels := required.PodLabels
+	namespace := "default"
+	ns := required.Metadata[AttrNamespace]
+	if ns != nil {
+		namespace = ns.re.String()
+	}
+
+	for attrName := range required.Metadata {
```

Comment:
Do we loop twice through the metadata because namespace and podname are allowed to be specified as labels and metadata?

Reply:
yes - that's the idea